File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(t) != len(s):
            return False
        seen = {}

        for char in s:
            if char in seen:
                seen[char] += 1
            else:
                seen[char] = 1
        
        for char in t:
            if char in seen:
                seen[char] -= 1
                if seen[char] == 0:
                    del seen[char]

        return len(seen) == 0

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();

        for (int i: nums) {
            if (seen.contains(i)) {
                return true;
            }
            else {
                seen.add(i);
            }
        }

        return false;
    }
}
